components/server/src/session-handler.ts

```diff
@@ -47,46 +45,83 @@ export class SessionHandlerProvider {
         options.store = this.createStore();
 
         this.sessionHandler = (req, res, next) => {
-            let hasJWTCookie = false;
-
             const cookies = parseCookieHeader(req.headers.cookie || "");
             const jwtToken = cookies[SessionHandlerProvider.getJWTCookieName(this.config)];
             if (jwtToken) {
                 // we handle the verification async, because we don't yet need to use it in the application
                 /* tslint:disable-next-line */
-                this.jwtSessionHandler(jwtToken)
-                    .then((res) => {
-                        hasJWTCookie = true;
-                    })
+                this.jwtSessionHandler(jwtToken, req, res, next)
+                    // the jwtSessionHandler is self-contained with respect to handling errors
+                    // however, if we did miss any, we at least capture it in here.
                     .catch((err) => {
-                        log.error("Failed to verify JWT Session token", err);
-                    })
-                    .finally(() => {
-                        reportSessionWithJWT(hasJWTCookie);
+                        log.error("Failed authenticate user with JWT Session cookie", err);
+                        res.statusCode = 500;
+                        res.send("Failed to authenticate jwt session.");
                     });
+            } else {
+                session(options)(req, res, next);
             }
-
-            session(options)(req, res, next);
         };
     }
 
-    protected async jwtSessionHandler(jwtToken: string): Promise<[JwtPayload, User]> {
-        const claims = await this.authJWT.verify(jwtToken);
-        log.debug("JWT Session token verified", {
-            claims,
-        });
+    protected async jwtSessionHandler(
+        jwtToken: string,
+        req: express.Request,
+        res: express.Response,
+        next: express.NextFunction,
+    ): Promise<void> {
+        try {
+            const claims = await this.authJWT.verify(jwtToken);
+            log.debug("JWT Session token verified", {
+                claims,
+            });
+
+            const subject = claims.sub;
+            if (!subject) {
+                throw new Error("Subject is missing from JWT session claims");
+            }
 
-        const subject = claims.sub;
-        if (!subject) {
-            throw new Error("Subject is missing from JWT session claims");
-        }
+            const user = await this.userDb.findUserById(subject);
```

Comment:
@easyCZ Outside of the scope for this PR, but mentioning bc I recall we were touching on this the other day: Note that this _is_ more costly than the websocket case, because after initiating a fresh websocket connection (every 10mins, or once per page load on the dashboard currently), requests no longer hit the HTTP handler chain.

Anyways, we can figure out caching another day and PR.